### ISSUE TITLE
Login With GOOGLE

### ISSUE THREAD

**Issue by joydeb28** [FusionAuth/fusionauth-issues#714]
## Login With GOOGLE
I was trying to login with GOOGLE and I noticed user.create webhook event that doesn't have the registration parameter whereas user.delete contains.
But in the source under user details, I am getting the details. Only its not coming in user.create.
![user create and delete](https://user-images.githubusercontent.com/25972361/86148922-e55e6780-bb18-11ea-85c8-442c83f75303.png)

**Comment by mooreds** [comment]
I think you might want the [user.registration.create](https://fusionauth.io/docs/v1/tech/events-webhooks/events#user-registration-create) event rather than (or perhaps in addition to) the user.create event.

I'm not sure what you mean by:

> in the source under user details, I am getting the details

Can you explain further?

**Comment by joydeb28** [comment]
I started both user.create and user.registration.create event but still getting the same response.

**Comment by mooreds** [comment]
So, just so I understand, the issue is that you are not getting the `registration` data in a webhook when a user signs up for a FusionAuth application using Google as the identity provider? And that you are have tried both the `user.create` and `user.registration.create` webhooks?

Is that correct?

**Comment by joydeb28** [comment]
Yes.

**Comment by mooreds** [comment]
So I just tried to recreate and I saw the `user.registration.create` event fire when I signed up with fusionauth that had a google identity provider configured. Here's what I saw in the webhook:

```
{
  event: {
    type: 'user.registration.create',
    applicationId: '73eda0dd-d559-4c5d-96bb-24a04cc4cc10',
    createInstant: 1593617345286,
    id: '2cf9ead3-8edc-4828-8e70-98a3349c0828',
    tenantId: '1de156c2-2daa-a285-0c59-b52f9106d4e4',
    user: {
      active: true,
      email: 'REDACTED',
      firstName: 'Dan',
      id: 'b3ed9346-09a8-40e8-860b-717188e254c8',
      insertInstant: 1591908857251,
      lastLoginInstant: 1593616224653,
      lastName: 'Moore',
      passwordChangeRequired: false,
      passwordLastUpdateInstant: 1591908857343,
      tenantId: '1de156c2-2daa-a285-0c59-b52f9106d4e4',
      twoFactorDelivery: 'None',
      twoFactorEnabled: false,
      usernameStatus: 'ACTIVE',
      verified: true
    }
  }
}
```

So from this, you basically have everything that the `user.delete` webhook gives you in the `registrations` object (`applicationId`, the `id` of the registration, `insertInstant`, `lastLoginInstant`). 

I guess you don't have all the applications you are registered for--is that what you are looking for? If so, I'd recommend catching this event in your webhook, and then looking up all the registrations for a given user id. The best way to do that is to [retrieve the user by their id](https://fusionauth.io/docs/v1/tech/apis/users#retrieve-a-user).

If I am misunderstanding what you are trying to do, please let me know.

**Comment by joydeb28** [comment]
I got your point.
But in the screenshot what I shared you can see for `delete webhook` registration data is coming.
But in `create webhook` it is not coming.
![user create and delete](https://user-images.githubusercontent.com/25972361/86431345-35544e80-bd12-11ea-8b99-2fb4f9b0f0af.png)

**Comment by mooreds** [comment]
Hiya.

Are you doing a user create or a user create + registration? 

If the former, there's no registration affiliated with the user at the time of the event, so that data won't be present.

If the latter, you'd have to catch the `user.registration.create` event, which should give you what you need.

Maybe I don't understand your use case; can you explain to me what you're trying to do?

**Comment by robotdan** [comment]
There may be some confusion on what will be in the events, I see the doc had some errors with the example JSON response. These are fixed now. https://fusionauth.io/docs/v1/tech/events-webhooks/events

User Create Event -> User only (no registrations)
```
{
  "type": "user.create",
  "user": {
  }
}
```

User Register Create -> User (no registrations) + New Registration
```
{
  "type": "user.registration.create",
  "registration": {
   }
  "user": {
  }
}
```

User Delete -> User (everything)
```
{
  "type": "user.delete",
  "user": {
     "registrations": []
  }
}
```

So when you log into Google for the first time, you'll see a `user.create` and a `user.registration.create` event.  The first event will only contain the user, the second event will contain the `user` and the `registration` top level objects in the event.

**Comment by joydeb28** [comment]
@robotdan 
I have enabled both `user.create` and `user.registration.create`.
But as I shared with you I am not getting the details of the registrations.

**Comment by robotdan** [comment]
Thanks for posting those events, that is helpful. I'll take a look at what is going on here.

**Comment by joydeb28** [comment]
@robotdan Thanks for taking it as an high priority.
Please let me know once it is available.

**Comment by robotdan** [comment]
Fix is available in 1.18.0.
